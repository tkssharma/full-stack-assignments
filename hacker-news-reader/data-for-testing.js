export const stories = [
  {
    by: 'magoghm',
    descendants: 41,
    id: 18277531,
    kids: [18277926, 18277900, 18278074, 18277699, 18277859, 18277835, 18277864],
    score: 79,
    time: 1540236259,
    title: 'AWS CEO Jassy follows Apple in calling for retraction of Chinese spy chip story',
    type: 'story',
    url:
      'https://www.cnbc.com/2018/10/22/aws-ceo-jassy-follows-apple-calls-for-spy-chip-story-retraction.html',
  },
  {
    by: 'NN88',
    descendants: 72,
    id: 18276862,
    kids: [
      18277851,
      18277773,
      18277336,
      18277638,
      18277523,
      18277196,
      18277173,
      18277302,
      18277409,
      18277700,
      18277870,
      18277390,
      18277646,
      18277581,
      18277721,
      18277324,
      18277689,
      18277514,
      18277389,
    ],
    score: 105,
    time: 1540231727,
    title: 'The Secretive Business of Facial-Recognition Software in Retail Stores',
    type: 'story',
    url:
      'http://nymag.com/intelligencer/2018/10/retailers-are-using-facial-recognition-technology-too.html',
  },
];

// we need time to be fixed relative to now (like 2 hours ago),
// we need the number of seconds since 1970.
// Note, because the props or the way the time is displayed always changes, we can't do snapshot on this component
var time = Date.now() * 1 - 1000 * 3600 * 3;
time = Math.round(time / 1000);

export const listItemData = {
  by: 'NN88',
  kids: [
    18277581,
    18277336,
    18277409,
    18277173,
    18277196,
    18277523,
    18277390,
    18277302,
    18277324,
    18277514,
    18277389,
  ],
  score: 58,
  url:
    'http://nymag.com/intelligencer/2018/10/retailers-are-using-facial-recognition-technology-too.html',
  title: 'The Secretive Business of Facial-Recognition Software in Retail Stores',
  id: 18276862,
  type: 'story',
  time: time,
};

export const gridItemData = {
  url:
    'http://nymag.com/intelligencer/2018/10/retailers-are-using-facial-recognition-technology-too.html',
  title: 'The Secretive Business of Facial-Recognition Software in Retail Stores',
  id: 18276862,
};

export const AppData = {
  hasMoreStores: false,
  isFetching: false,
  layout: 'list',
  page: 1,
  stories: stories,
  storyIds: [18277531, 18276862],
  theme: 'dark',
};

export const NavData = {
  layout: 'list',
  theme: 'dark',
};
