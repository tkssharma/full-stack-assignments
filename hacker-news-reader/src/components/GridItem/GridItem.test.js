import React from 'react';
import { gridItemData } from '../../../data-for-testing';
import { shallow } from 'enzyme';
import GridItem from './index';
const { url, title, id } = gridItemData;

it('renders without crashing', () => {
  const component = shallow(<GridItem url={url} title={title} id={id} />);
  expect(component).toMatchSnapshot();
});
