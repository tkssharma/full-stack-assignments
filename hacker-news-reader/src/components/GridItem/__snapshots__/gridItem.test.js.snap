// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GridItem
    id={18276862}
    title="The Secretive Business of Facial-Recognition Software in Retail Stores"
    url="http://nymag.com/intelligencer/2018/10/retailers-are-using-facial-recognition-technology-too.html"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <styled.li>
        <styled.div>
                <styled.img
                        src="https://miro.medium.com/max/1176/1*F9RzuXseG1VrTjFJd403gw.png"
                />
                <styled.div>
                        <styled.h3>
                                The Secretive Business of Facial-Recognition Software in Retail Stores
                        </styled.h3>
                        <styled.div>
                                // 
                                nymag.com
                        </styled.div>
                </styled.div>
        </styled.div>
</styled.li>,
      "href": "http://nymag.com/intelligencer/2018/10/retailers-are-using-facial-recognition-technology-too.html",
      "rel": "nofollow noreferrer nofollow",
      "target": "_blank",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.img
                    src="https://miro.medium.com/max/1176/1*F9RzuXseG1VrTjFJd403gw.png"
          />
          <styled.div>
                    <styled.h3>
                              The Secretive Business of Facial-Recognition Software in Retail Stores
                    </styled.h3>
                    <styled.div>
                              // 
                              nymag.com
                    </styled.div>
          </styled.div>
</styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.img
              src="https://miro.medium.com/max/1176/1*F9RzuXseG1VrTjFJd403gw.png"
/>,
            <styled.div>
              <styled.h3>
                            The Secretive Business of Facial-Recognition Software in Retail Stores
              </styled.h3>
              <styled.div>
                            // 
                            nymag.com
              </styled.div>
</styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "src": "https://miro.medium.com/max/1176/1*F9RzuXseG1VrTjFJd403gw.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.h3>
                  The Secretive Business of Facial-Recognition Software in Retail Stores
</styled.h3>,
                <styled.div>
                  // 
                  nymag.com
</styled.div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "The Secretive Business of Facial-Recognition Software in Retail Stores",
                },
                "ref": null,
                "rendered": "The Secretive Business of Facial-Recognition Software in Retail Stores",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "// ",
                    "nymag.com",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "// ",
                  "nymag.com",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <styled.li>
          <styled.div>
                    <styled.img
                              src="https://miro.medium.com/max/1176/1*F9RzuXseG1VrTjFJd403gw.png"
                    />
                    <styled.div>
                              <styled.h3>
                                        The Secretive Business of Facial-Recognition Software in Retail Stores
                              </styled.h3>
                              <styled.div>
                                        // 
                                        nymag.com
                              </styled.div>
                    </styled.div>
          </styled.div>
</styled.li>,
        "href": "http://nymag.com/intelligencer/2018/10/retailers-are-using-facial-recognition-technology-too.html",
        "rel": "nofollow noreferrer nofollow",
        "target": "_blank",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div>
            <styled.img
                        src="https://miro.medium.com/max/1176/1*F9RzuXseG1VrTjFJd403gw.png"
            />
            <styled.div>
                        <styled.h3>
                                    The Secretive Business of Facial-Recognition Software in Retail Stores
                        </styled.h3>
                        <styled.div>
                                    // 
                                    nymag.com
                        </styled.div>
            </styled.div>
</styled.div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.img
                src="https://miro.medium.com/max/1176/1*F9RzuXseG1VrTjFJd403gw.png"
/>,
              <styled.div>
                <styled.h3>
                                The Secretive Business of Facial-Recognition Software in Retail Stores
                </styled.h3>
                <styled.div>
                                // 
                                nymag.com
                </styled.div>
</styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "src": "https://miro.medium.com/max/1176/1*F9RzuXseG1VrTjFJd403gw.png",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.h3>
                    The Secretive Business of Facial-Recognition Software in Retail Stores
</styled.h3>,
                  <styled.div>
                    // 
                    nymag.com
</styled.div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "The Secretive Business of Facial-Recognition Software in Retail Stores",
                  },
                  "ref": null,
                  "rendered": "The Secretive Business of Facial-Recognition Software in Retail Stores",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "// ",
                      "nymag.com",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "// ",
                    "nymag.com",
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
